pkg/service/v0/service.go

```diff
@@ -64,32 +71,85 @@ func (g Service) SaveBundle(c context.Context, req *proto.SaveBundleRequest, res
 
 // GetBundle implements the BundleServiceHandler interface
 func (g Service) GetBundle(c context.Context, req *proto.GetBundleRequest, res *proto.GetBundleResponse) error {
+	accountUUID := getValidatedAccountUUID(c, "me")
 	if validationError := validateGetBundle(req); validationError != nil {
 		return merrors.BadRequest("ocis-settings", "%s", validationError)
 	}
-	r, err := g.manager.ReadBundle(req.BundleId)
+	bundle, err := g.manager.ReadBundle(req.BundleId)
 	if err != nil {
 		return merrors.NotFound("ocis-settings", "%s", err)
 	}
-	res.Bundle = r
+	roleIDs := g.getRoleIDs(c, accountUUID)
+	filteredBundle := g.getFilteredBundle(roleIDs, bundle)
+	if len(filteredBundle.Settings) == 0 {
+		err = fmt.Errorf("could not read bundle: %s", req.BundleId)
+		return merrors.NotFound("ocis-settings", "%s", err)
+	}
+	res.Bundle = filteredBundle
 	return nil
 }
 
 // ListBundles implements the BundleServiceHandler interface
 func (g Service) ListBundles(c context.Context, req *proto.ListBundlesRequest, res *proto.ListBundlesResponse) error {
 	// fetch all bundles
-	req.AccountUuid = getValidatedAccountUUID(c, req.AccountUuid)
+	accountUUID := getValidatedAccountUUID(c, "me")
```

Comment:
Why the call to `getValidatedAccountUUID` if the result is always an empty string because of the parameter `"me"`?

Reply:
We want to have the account uuid of the authenticated user. See comment on your other comment. :)